cmd/proxy/proxy.go

```diff
@@ -279,6 +180,62 @@ func (c *Command) initEventBase() {
 	}
 }
 
+func (c *Command) populateConfig() error {
+	if c.flags.config == "" {
+		return nil
+	}
+	b, err := os.ReadFile(c.flags.config)
+	if err != nil {
+		return fmt.Errorf("read config file: %w", err)
+	}
+
+	var config config
```

Comment:
just parse things into `c.config` directly. if this function returns an error, the process isn't going to ignore it, so `c.config` being partially populated isn't a problem.